src/Redux/rockets/rocketSlice.js

```diff
@@ -0,0 +1,80 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const url = 'https://api.spacexdata.com/v3/rockets';
+
+const initialState = {
+  rockets: [],
+  isLoading: true,
+  error: undefined,
+};
+
+export const getResultItems = createAsyncThunk('result/getResultItems', async (thunkAPI) => {
+  try {
+    const resp = await axios(url);
+    const myData = resp.data;
+
+    const myTransData = [];
+
+    Object.keys(myData).forEach((key) => {
+      const myTmpObj = {
+        id: myData[key].id,
+        imgPath: myData[key].flickr_images[0],
+        rocketName: myData[key].rocket_name,
+        active: myData[key].active,
+        description: myData[key].description,
+      };
+      myTransData.push(myTmpObj);
+    });
+
+    return myTransData;
+  } catch (error) {
+    return thunkAPI.rejectWithValue('something went wrong...');
+  }
+});
+
+const rocketSlice = createSlice({
+  name: 'rocketHandler',
+  initialState,
+  reducers: {
+    reserveRocket: (state, action) => {
+      const rocketId = action.payload.id;
+      const ansRockets = [];
+      state.rockets.forEach((rocket) => {
+        const tmpRockt = rocket;
+        if (tmpRockt.id === rocketId) {
+          tmpRockt.active = false;
+        }
+        ansRockets.push(tmpRockt);
+      });
+      state.rockets = ansRockets;
```

Comment:
- [ ] I will encourage you to double-check the logic of this active toggling, when I run the app, it by default makes the first falcon and the starship reserved by default which should not be since I have not reserved them.

![image](https://github.com/78fatemanazari/react-capstone/assets/86200356/d553fd84-4ef0-4975-bf19-46400729eac4)